collector/btrfs_linux_test.go

```diff
@@ -0,0 +1,118 @@
+		{name: "reserved_bytes", value: 0, extraLabel: []string{"type"}, extraLabelValue: []string{"system"}},
+		{name: "used_bytes", value: 16384, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"system", "raid1"}},
+		{name: "total_bytes", value: 8.388608e+06, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"system", "raid1"}},
+		{name: "ratio", value: 2, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"system", "raid1"}},
+	},
+	{
+		{name: "devices", value: 4},
+		{name: "global_rsv_size_bytes", value: 1.6777216e+07},
+		{name: "device_size", value: 1.073741824e+10, extraLabel: []string{"device"}, extraLabelValue: []string{"loop22"}},
+		{name: "device_size", value: 1.073741824e+10, extraLabel: []string{"device"}, extraLabelValue: []string{"loop23"}},
+		{name: "device_size", value: 1.073741824e+10, extraLabel: []string{"device"}, extraLabelValue: []string{"loop24"}},
+		{name: "device_size", value: 1.073741824e+10, extraLabel: []string{"device"}, extraLabelValue: []string{"loop25"}},
+		{name: "reserved_bytes", value: 0, extraLabel: []string{"type"}, extraLabelValue: []string{"data"}},
+		{name: "used_bytes", value: 0, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"data", "raid5"}},
+		{name: "total_bytes", value: 6.44087808e+08, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"data", "raid5"}},
+		{name: "ratio", value: 1.3333333333333333, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"data", "raid5"}},
+		{name: "reserved_bytes", value: 0, extraLabel: []string{"type"}, extraLabelValue: []string{"metadata"}},
+		{name: "used_bytes", value: 114688, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"metadata", "raid6"}},
+		{name: "total_bytes", value: 4.29391872e+08, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"metadata", "raid6"}},
+		{name: "ratio", value: 2, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"metadata", "raid6"}},
+		{name: "reserved_bytes", value: 0, extraLabel: []string{"type"}, extraLabelValue: []string{"system"}},
+		{name: "used_bytes", value: 16384, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"system", "raid6"}},
+		{name: "total_bytes", value: 1.6777216e+07, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"system", "raid6"}},
+		{name: "ratio", value: 2, extraLabel: []string{"type", "mode"}, extraLabelValue: []string{"system", "raid6"}},
+	},
+}
+
+func checkMetric(exp, got *btrfsMetric) bool {
+	if exp.name != got.name ||
+		exp.value != got.value ||
+		len(exp.extraLabel) != len(got.extraLabel) ||
+		len(exp.extraLabelValue) != len(got.extraLabelValue) {
+		return false
+	}
+
+	for i := range exp.extraLabel {
+		if exp.extraLabel[i] != got.extraLabel[i] {
+			return false
+		}
+
+		// Devices (loopXX) can appear in random order, so just check the first 4 character
```

Comment:
All comments must start with a capitol and end with a period.

```suggestion
		// Devices (loopXX) can appear in random order, so just check the first 4 character.
```

Reply:
And also plural :disappointed: